### ISSUE TITLE
experiments: weakly supervised learning of senses

### ISSUE THREAD

**Issue by BarbaraMcG** [Living-with-machines/TargetedSenseDisambiguation#57]
This was inspired by https://github.com/Living-with-machines/HistoricalDictionaryExpansion/issues/39 and specifically this paper: 
SenseBERT: Driving Some Sense into BERT, Yoav Levine, Barak Lenz, Or Dagan, Ori Ram, Dan Padnos, Or Sharir, Shai Shalev-Shwartz, Amnon Shashua, Yoav Shoham
https://www.aclweb.org/anthology/2020.acl-main.423/

@kasparvonbeelen 's summary:
 
What is this paper about?
The paper demonstrates how self-supervision can infuse BERT with richer semantic representations. It is focussed on the pre-training phase. Besides the traditional masked language modelling, the authors train BERT to predict the WordNet supersense of a masked token. Even a token may map to different supersenses, the context will fit some supersenses better than others, e.g. words appearing in the sentence this is [MASK] delicious will likely be of category noun.food. This proxy or soft-labelling technique allows for self-supervision.

The authors show the SenseBERT representation capture and improve performance on downstream tasks.

Is it relevant to our project? If so, why and how?
Yes, we could extend/adapt the method to OED thesaurus, and attempt the predict parents of the semantic class of a sense.

for the quotations, we know the "correct" parents (high-quality data)
for other words (in quotations or any other corpus), we can use soft labelling as proposed by the authors.
We could assess if this type of pre-training improves the WSD tasks, or generate better vector representation.

More interestingly even, we could improve pretraining by adapting the sentence level predictions procedure, see if BERT can predict aspects of a sentence's metadata (e.g. year and author). Now BERT only attempts to predict the next sentence.
